src/hotspot/share/oops/instanceKlass.cpp

```diff
@@ -955,7 +955,7 @@ bool InstanceKlass::link_class_impl(TRAPS) {
 
   // Could it be possible to do the following processing only if the
   // class uses inline types?
-  if (EnableValhalla) {
+  if (EnablePrimitiveClasses) {
```

Comment:
This actually breaks the following test case:

build/vscode-slowdebug/jdk/bin/java -XX:+EnableValhalla -XX:-EnablePrimitiveClasses -Xlog:class+preload ValuePreloadClient0

Preload won't occur